appserver/common/container-common/src/main/java/com/sun/enterprise/container/common/impl/managedbean/ManagedBeanManagerImpl.java

```diff
@@ -549,17 +541,15 @@ private boolean bundleEligible(BundleDescriptor bundle) {
 
             // Have 299 create, inject, and call PostConstruct (if desired) on managed bean
 
-            JCDIService.JCDIInjectionContext jcdiContext =
+            JCDIService.JCDIInjectionContext<?> jcdiContext =
                 jcdiService.createManagedObject(managedBeanClass, bundleDescriptor, invokePostConstruct);
             callerObject = (T) jcdiContext.getInstance();
             // Need to keep track of context in order to destroy properly
-            Map<Object, JCDIService.JCDIInjectionContext> bundleNonManagedObjs =
+            Map<Object, JCDIService.JCDIInjectionContext<?>> bundleNonManagedObjs =
                 jcdiManagedBeanInstanceMap.get(bundleDescriptor);
             bundleNonManagedObjs.put(callerObject, jcdiContext);
 
-        } else {
-
-
+        } else if (desc != null) {
```

Comment:
In such case, and same exception thrown here if `desc` is null

Isn't there an opportunity to extract common parts of these methods?

Reply:
I changed to throwing the exception you suggested.
I also looked into the duplication and from the method signatures I'd assume those should be almost the same but there are strange differences in the details. It might just be that code paths were not updated properly but I felt this is too big of a change to fit into the scope of this PR so I left it as is.